tamr_unify_client/models/dataset/resource.py

```diff
@@ -66,20 +68,60 @@ def update_records(self, records, **json_args):
         :rtype: :py:class:`dict`
         """
 
-        def _stringify_updates(updates):
-            for update in updates:
+        def _stringify_updates(updaters):
+            for update in updaters:
                 yield json.dumps(update, **json_args).encode("utf-8")
 
         return (
             self.client.post(
                 self.api_path + ":updateRecords",
                 headers={"Content-Encoding": "utf-8"},
-                data=_stringify_updates(records),
+                data=_stringify_updates(updates),
             )
             .successful()
             .json()
         )
 
+    def upsert_records(self, records, key_name, **json_args):
+        """Creates or updates the specified records.
+
+        :param records: The records to update, as dictionaries.
+        :type records: iterable[dict]
+        :param key_name: The name of the primary key for these records, which must be a key in each record dictionary.
+        :type key_name: str
+        :param `**json_args`: Arguments to pass to the JSON `dumps` function, as documented `here <https://simplejson.readthedocs.io/en/latest/#simplejson.dumps>`_.
+            Some of these, such as `indent`, may not work with Unify.
+        :return: JSON response body from the server.
+        :rtype: dict
+        """
+
+        def _to_updates(updates):
```

Comment:
Can simplify via:

```python
updates = (
  {
    "action": "CREATE",
    "recordId": record[primary_key],
    "record": record,
  } for record in records
)

return self._update_records(updates, **json_args)
```

Reply:
Even if you don't go with this^, the argument should be renamed from `updates` to `records`:

```python
def _to_updates(records):
```